src/apps/vpn/platforms/android/androidcontroller.cpp

```diff
@@ -106,56 +106,64 @@ AndroidController::~AndroidController() { MZ_COUNT_DTOR(AndroidController); }
 void AndroidController::initialize(const Device* device, const Keys* keys) {
   logger.debug() << "Initializing";
 
-  Q_UNUSED(device);
   Q_UNUSED(keys);
 
+  m_deviceName = device->name();
+  m_devicePublicKey = device->publicKey();
+  m_deviceCreationTime = device->createdAt().toMSecsSinceEpoch();
+
   AndroidVPNActivity::connectService();
 }
 
-void AndroidController::activate(const HopConnection& hop, const Device* device,
-                                 const Keys* keys, Controller::Reason reason) {
-  Q_ASSERT(hop.m_hopindex == 0);
+void AndroidController::activate(const InterfaceConfig& config,
+                                 Controller::Reason reason) {
+  Q_ASSERT(config.m_hopType == InterfaceConfig::SingleHop);
```

Comment:
We should consider also just doing `QJsonObject json = config.toJson();`
and adding the few extra things android requires to that obj. 
However that can be a followup that i gladly will take.

Reply:
I thought about it, but the format of the JSON used by the `AndroidController` is slightly different from what the desktop platforms use, so I opted to leave this method mostly untouched.